src/Sentry/Platforms/Cocoa/CocoaProfiler.cs

```diff
@@ -36,8 +36,7 @@ public void Finish()
 
     public ISerializable? Collect(Transaction transaction)
     {
-        // TODO change return type of CocoaSDKs CollectProfileBetween to NSMutableDictionary
-        var payload = SentryCocoaHybridSdk.CollectProfileBetween(_startTimeNs, _endTimeNs, _cocoaTraceId)?.MutableCopy() as NSMutableDictionary;
+        var payload = SentryCocoaHybridSdk.CollectProfileBetween(_startTimeNs, _endTimeNs, _cocoaTraceId) as NSMutableDictionary;
```

Comment:
Is the purpose of this PR just to remove the TODO/comment?

Reply:
and `?.MutableCopy()`

Reply:
and actually, the cast is also unnecessary, I think.
removing now, let's see what the CI says